### PR TITLE
Migrate psammead-episode-list to Simorgh

### PR DESCRIPTION
**Pull request by RichardPK**
Ongoing audit of Psammead packages that should be considered for moving to Simorgh, here > https://paper.dropbox.com/doc/Psammead-Simorgh-package-audit--BGKHUPYrS3lbnhV9Uy9dUybVAQ-m4oP7K0VcpJpajcT1LIJK

## Why

- We have discussed the need and desire to move large, composite, Simorgh-specific components out of Psammead and into Simorgh to:
1. Reduce the number of packages in Psammead that import multiple other Psammead packages
2. Reduce the code admin workload inherently connected to maintaining Psammead components that are opinionated and likely only useful to Simorgh
3. Reducing the number of Psammead packages also reduces the work required whenever a change needs propagating to every Psammead package (i.e. an Emotion / Storybook major bump)

- As PRs are better at triggering conversations than docs, this is an idea for how we might migrate a package like [`psammead-episode-list`](https://github.com/bbc/psammead/tree/4e197c32e1ebb6c8e5b2749216cc9e75f561dd30/packages/components/psammead-episode-list)

## Code Changes

- Move the `psammead-episode-list` `/src` folder into `Containers/EpisodeList`
- Move `RecentAudioEpisodes` and `RecentVideoEpisodes` into that folder.
- Create `.stories.jsx` files within both of those folders, using these two components & add localised component fixtures.
- Remove various psammead examples & stories - https://github.com/bbc/simorgh/pull/8879/commits/3c99be68aab735325b8628952394e734621133b5 
- Rewrite `.index.test.jsx` to make use of `RecentAudioEpisodes` & `RecentVideoEpisodes` components.
New component structure:
Screenshot 2021-03-01 at 09 34 44

## Questions
- Should the 'core' components, i.e. `DateTimeDuration`, `Description`, `Title` etc. be within their own folder? Should the testing of these be totally separated from the `RecentVideoEpisodes` / `RecentAudioEpisodes` components?
- Storybook. Is it OK that these 'core' components are not Storybooked in isolation - and instead Storybook is only used for the 'composite' components - `RecentVideoEpisodes` & `RecentAudioEpisodes`

## Notes

Two follow up issues:
- Add `recentEpisodeLists` integration tests to OD-TV & OD-radio's page level integration tests.
- Support our colour pallet in Storybook dark mode - currently too light.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/EpisodeList/DateTimeDuration.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    ? `  
+    padding-left: ${GEL_SPACING};
+    margin-left: ${GEL_SPACING};
+    border-left: 0.0625rem solid ${C_CLOUD_LIGHT};`
+    : `
+    padding-right: ${GEL_SPACING};
+    margin-right: ${GEL_SPACING};
+    border-right: 0.0625rem solid ${C_CLOUD_LIGHT};`
+}
+`;
```
Comment (ryanmccombe) [inline src/app/containers/EpisodeList/DateTimeDuration.jsx:21]:
```suggestion
border-left: 0.0625rem solid ${C_CLOUD_LIGHT};
${
  dir === 'ltr'
    ? `  
    padding-left: ${GEL_SPACING};
    margin-left: ${GEL_SPACING};`
    : `
    padding-right: ${GEL_SPACING};
    margin-right: ${GEL_SPACING};`
}
`;
```

**Review by ryanmccombe — APPROVED** [review]

File: src/app/pages/OnDemandAudioPage/OnDemandAudioPage.jsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ import getMediaId from '#lib/utilities/getMediaId';
 import getMasterbrand from '#lib/utilities/getMasterbrand';
 import getEmbedUrl from '#lib/utilities/getUrlHelpers/getEmbedUrl';
 import RadioScheduleContainer from '#containers/RadioSchedule';
-import RecentAudioEpisodes from '#containers/RecentAudioEpisodes';
+import RecentAudioEpisodes from '#app/containers/EpisodeList/RecentAudioEpisodes';
```
Comment (rebeccamcginn) [inline src/app/pages/OnDemandAudioPage/OnDemandAudioPage.jsx]:
```suggestion
import RecentAudioEpisodes from '#containers/EpisodeList/RecentAudioEpisodes';
```

File: src/app/pages/OnDemandAudioPage/OnDemandAudioPage.jsx [hunk 3]
```diff
@@ -26,7 +26,7 @@ import getMediaId from '#lib/utilities/getMediaId';
...
 import FooterTimestamp from '#app/containers/OnDemandFooterTimestamp';
```
Comment (rebeccamcginn) [inline src/app/pages/OnDemandAudioPage/OnDemandAudioPage.jsx]:
```suggestion
import FooterTimestamp from '#containers/OnDemandFooterTimestamp';
```

**Review by rebeccamcginn — APPROVED** [review]
Looks good! I feel it makes sense to keep the core components within `Containers/EpisodeList`

**Comment by rebeccamcginn** [comment]
Test-wise, `EpisodeList/index.test.jsx` is audio-heavy, with only two tests for video... I'd be tempted to move the tests (excluding any duplicates) into `RecentAudioEpisodes/index.test.jsx` & `RecentVideoEpisodes/index.test.jsx`.

File: src/app/pages/OnDemandAudioPage/OnDemandAudioPage.jsx (hunk 2 above)
Comment (RichardPK) [reply src/app/pages/OnDemandAudioPage/OnDemandAudioPage.jsx]: Ty!

**Comment by RichardPK** [comment]
> Test-wise, EpisodeList/index.test.jsx is audio-heavy, with only two tests for video... I'd be tempted to move the tests (excluding any duplicates) into RecentAudioEpisodes/index.test.jsx & RecentVideoEpisodes/index.test.jsx.

Yea I guess why I kept that one @rebeccamcginn is with the idea that there would be _some_ separation of responsibility between the simple/core styled components, and then when those things are composed.

That being said... I'm just relentlessly using the RecentEpisode composite components, as you've highlighted, for these tests... so it doesn't make sense. 

I'll check tomorrow if any of those styled comps warrant their own tests, and I'll leave them in. Any others I'll move into the respective composite component tests

**Comment by RichardPK** [comment]
My gut is that actually most of the tests, especially those in `src/app/containers/EpisodeList/index.test.jsx` like https://github.com/bbc/simorgh/pull/8879/files#diff-e32780b4a5bd0a4ffc702a6fef929e34249e958ce4a8dbf6e329e6bb04fcc818R68 should be a set of integration tests. I think a number of the `RecentVideoEpisodes` & `RecentAudioEpisodes` tests could also be moved into integration tests too.

@jroebu14 - interested to know what you think?

**Comment by chris-hinds** [comment]
Cheers @RichardPK this PR looks good.

Have we put any thought into what we do with the Psammead package after this migration?

**Review by chris-hinds — APPROVED** [review]

**Comment by RichardPK** [comment]
Merged @chris-hinds - I'll thread a discussion on the steps to help figure out how to decomm a Psammead package?

**Pull request merged**